src/NodeRegistry.sol

```diff
@@ -102,134 +97,109 @@ contract NodeRegistry is INodeRegistry, AccessControlDefaultAdminRules, ERC721 {
 
         nodeId = ++_nodeCounter * NODE_INCREMENT; // The first node starts with `nodeId = NODE_INCREMENT`.
 
-        _nodes[nodeId] = Node(signingKeyPub, httpAddress, false, false, false, minMonthlyFeeMicroDollars);
+        _nodes[nodeId] = Node(signingKeyPub, httpAddress, false, minMonthlyFeeMicroDollars);
 
         _mint(to, nodeId);
 
         emit NodeAdded(nodeId, to, signingKeyPub, httpAddress, minMonthlyFeeMicroDollars);
     }
 
-    /// @inheritdoc INodeRegistry
-    function disableNode(uint256 nodeId) public onlyRole(ADMIN_ROLE) {
+    /**
+     * @inheritdoc INodeRegistry
+     */
+    function addToNetwork(uint256 nodeId) external onlyRole(ADMIN_ROLE) {
         _revertIfNodeDoesNotExist(nodeId);
 
-        _nodes[nodeId].isDisabled = true;
+        require(_canonicalNetworkNodes.length() < _maxActiveNodes, MaxActiveNodesReached());
+        require(!_canonicalNetworkNodes.contains(nodeId), NodeAlreadyInCanonicalNetwork());
 
-        // Always remove from active nodes sets when disabled.
-        _disableApiNode(nodeId);
-        _disableReplicationNode(nodeId);
+        require(_canonicalNetworkNodes.add(nodeId), FailedToAddNodeToCanonicalNetwork());
```

Comment:
Seems there checks are redundant. A failure to add to the enumerated set would be ebcusae the set already contains the node.